src/app/lib/transaction.ts

```diff
@@ -79,6 +91,82 @@ export class OasisTransaction {
     return tw
   }
 
+  public static async buildStakingAllowTransfer(
+    nic: OasisClient,
+    signer: Signer,
+    to: string,
+    amount: bigint,
+  ): Promise<TW<oasis.types.StakingAllow>> {
+    const tw = oasis.staking.allowWrapper()
+    const nonce = await OasisTransaction.getNonce(nic, signer)
+    const beneficiary = await addressToPublicKey(to)
+
+    tw.setNonce(nonce)
+    tw.setFeeAmount(oasis.quantity.fromBigInt(0n))
+    tw.setBody({
+      beneficiary,
+      negative: false,
+      amount_change: oasis.quantity.fromBigInt(amount),
+    })
+
+    const gas = await tw.estimateGas(nic, signer.public())
+    tw.setFeeGas(gas)
+
+    return tw
+  }
+
+  public static async buildParatimeTransfer(
+    nic: OasisClient,
+    signer: Signer,
+    transaction: ParaTimeTransaction, // for amount other methods use bigint
+    fromAddress: string,
+    runtimeId: string,
+    runtimeDecimals: number,
+  ): Promise<RTW<oasisRT.types.ConsensusDeposit | oasisRT.types.ConsensusWithdraw>> {
+    const { amount, recipient: targetAddress, type } = transaction
+    const isDepositing = type === TransactionTypes.Deposit
+    const consensusRuntimeId = oasis.misc.fromHex(runtimeId)
+    const txWrapper = new oasisRT.consensusAccounts.Wrapper(consensusRuntimeId)[
+      isDepositing ? 'callDeposit' : 'callWithdraw'
+    ]()
+    const accountsWrapper = new oasisRT.accounts.Wrapper(consensusRuntimeId)
+    const nonce = await accountsWrapper
+      .queryNonce()
+      .setArgs({ address: await oasis.staking.addressFromBech32(fromAddress) })
+      .query(nic)
+    const feeAmountDecimal = new BigNumber(10).pow(consensusDecimals)
+    const feeAmount = BigInt(
+      new BigNumber(transaction.feeAmount ? transaction.feeAmount : isDepositing ? '0' : '1500000')
+        .shiftedBy(runtimeDecimals)
+        .dividedBy(feeAmountDecimal)
+        .toFixed(),
```

Comment:
See https://github.com/oasisprotocol/oasis-wallet-web/pull/969/files#r944765518
should be toFixed(0)